src/nanoemoji/nanoemoji.py

```diff
@@ -24,497 +24,135 @@
-    make_svg_table(ttfont, color_glyphs, picosvg, compressed)
-
-
-def _generate_fea(rgi_sequences):
-    # TODO if this is a qualified sequence create the unqualified version and vice versa
-    rules = []
-    rules.append("languagesystem DFLT dflt;")
-    rules.append("languagesystem latn dflt;")
-
-    rules.append("feature rlig {")
-    for rgi, target in sorted(rgi_sequences):
-        if len(rgi) == 1:
-            continue
-        glyphs = [glyph_name(cp) for cp in rgi]
-        rules.append("  sub %s by %s;" % (" ".join(glyphs), target))
-
-    rules.append("} rlig;")
-    return "\n".join(rules)
-
-
-def _ensure_codepoints_will_have_glyphs(ufo, glyph_inputs):
-    """Ensure all codepoints we use will have a glyph.
-
-    Single codepoint sequences will directly mapped to their glyphs.
-    We need to add a glyph for any codepoint that is only used in a multi-codepoint sequence.
-
-    """
-    all_codepoints = set()
-    direct_mapped_codepoints = set()
-    for _, codepoints, _ in glyph_inputs:
-        if len(codepoints) == 1:
-            direct_mapped_codepoints.update(codepoints)
-        all_codepoints.update(codepoints)
-
-    need_blanks = all_codepoints - direct_mapped_codepoints
-    logging.debug("%d codepoints require blanks", len(need_blanks))
-    glyph_names = []
-    for codepoint in need_blanks:
-        # Any layer is fine; we aren't going to draw
-        glyph = ufo.newGlyph(glyph_name(codepoint))
-        glyph.unicode = codepoint
-        glyph_names.append(glyph.name)
-
-    ufo.glyphOrder = ufo.glyphOrder + sorted(glyph_names)
-
-
-def _generate_color_font(config: ColorFontConfig, inputs: Iterable[InputGlyph]):
-    """Make a UFO and optionally a TTFont from svgs."""
-    ufo = _ufo(config.family, config.upem, config.keep_glyph_names)
-    _ensure_codepoints_will_have_glyphs(ufo, inputs)
-
-    base_gid = len(ufo.glyphOrder)
-    color_glyphs = [
-        ColorGlyph.create(ufo, filename, base_gid + idx, codepoints, psvg)
-        for idx, (filename, codepoints, psvg) in enumerate(inputs)
-    ]
-    ufo.glyphOrder = ufo.glyphOrder + [g.glyph_name for g in color_glyphs]
-    for g in color_glyphs:
-        assert g.glyph_id == ufo.glyphOrder.index(g.glyph_name)
-
-    _COLOR_FORMAT_GENERATORS[config.color_format].apply_ufo(ufo, color_glyphs)
-
-    ufo.features.text = _generate_fea(
-        [(c.codepoints, c.glyph_name) for c in color_glyphs]
-    )
-    logging.debug("fea:\n%s\n" % ufo.features.text)
 
-    ttfont = _make_ttfont(config, ufo, color_glyphs)
+def write_fea_build(nw: ninja_syntax.Writer, svg_files: Sequence[str]):
+    nw.build("features.fea", "write_fea", "codepointmap.csv")
+    nw.newline()
 
-    # TODO may wish to nuke 'post' glyph names
 
-    return ufo, ttfont
+def write_font_build(nw: ninja_syntax.Writer, svg_files: Sequence[str]):
+    picosvgs = [picosvg(f) for f in svg_files]
+    nw.build(
+        write_font.output_file(FLAGS.family, FLAGS.output, FLAGS.color_format),
```

Comment:
shouldn't the build graph reflect the fact that if ether the codepointmap or the features.fea change, the font should be rebuilt?
this dependency is sort of implicit right now, since the codepoint map and the features derive directly from the filenames of the input svgs. If we allow the user to pass their own codepointmap or features.fea, then write_font should take those as dependencies.